File: src/string/2810.js
/**
 * @param {string} s
 * @return {string}
 */
var finalString = function (s) {
  const result = [];

  for (let i = 0; i < s.length; i++) {
    if (s[i] === 'i') {
      result.reverse();
    } else {
      result.push(s[i]);
    }
  }

  return result.join('');
};

File: src/string/2810.cpp
class Solution {
public:
  string finalString(string s) {
    vector<char> arr;
    string str;
    
    for(int i=0; i<s.size(); i++)
    {
      if (s[i] == 'i')
      {
        reverse(arr.begin(), arr.end());
      } else
      {
        arr.push_back(s[i]);
      }
    }
    
    str.assign(arr.begin(), arr.end());
    return str;
  }
};
